content/md/posts/clojure-which-execution-option-to-use.md

```diff
@@ -0,0 +1,217 @@
+{:title "Clojure CLI tools - which execution option to use"
+ :layout :post
+ :date "2021-12-20"
+ :topic "clojure-cli"
+ :tags  ["clojure-cli" "tools-deps"]}
+
+Clojure CLI tools provide a very flexible way to run Clojure, using aliases to include community libraries and tools to enhance clojure projects and provide independent tools. Understand the execution options (exec-opts) on the command line options ensures an effective use of Clojure CLI tools.
+
+Clojure CLI tools was [first released in February 2020](https://clojure.org/releases/tools#v1.10.1.510) with a specific focus on running Clojure code quickly and easily, which also required the management of dependencies from Maven, Git and a local file space.
+
+Since that release the design has evolved to provide clojure.exec (`-X`) to run any Clojure function not just `-main` as with clojure.main (`-M`). In July 2021, the ability to run tools (`-T`) which are independent from the Clojure project classpath was also introduced.
+
+The exec-opts command line flags have evolved to enable these features, so lets explore those flags and show how each flag is typically used.
+
+<!-- more -->
+
+## Summary of use
+
+In very simple terms, the Clojure CLI tools flags are used as follows
+
+`-A` - [Deprecated] use `-M` instead
```

Comment:
`-A` is deprecated for running `:main-opts` but it is still the correct way to start a REPL and will continue to be so.